### PR TITLE
Added top level $ref suppport in JSONSchemaBridge.

### LINKED ISSUE #507: JSON Schema - Multiple Definitions Errors Out
When creating a `JSONSchemaBridge` of [this schema](https://github.com/DacioRomero/Stands-Schemas/blob/66879e2e4ffc796cee33769168f738c4da730c4d/2019.json) it errors when inputted into `AutoForm` and `ValidatedForm`

My initial impressions is that `JSONSchemaBridge` isn't expecting JSON Schema in the format that I used which I believe is standard.

Apologies if my React code is a bit unconventional; I'm new.

## Code and Logs

<details>

<summary>Source (omitted irrelevant parts)</summary>

```javascript
import Ajv from 'ajv';
import JSONSchemaBridge from 'uniforms/JSONSchemaBridge';
import AutoForm from 'uniforms/AutoForm';


export default class extends Component {
  state = { schemaBridge: null };
  
  async componentDidMount() {
    const response = await fetch('https://gitcdn.xyz/repo/DacioRomero/Stands-Schemas/master/2019.json');
    const schema = await response.json();

    const validator = new Ajv({ allErrors: true, useDefaults: true }).compile(schema);

    const schemaValidator = model => {
        validator(model);

        if (validator.errors && validator.errors.length) {
            throw {details: validator.errors};
        }
    };

    const schemaBridge = new JSONSchemaBridge(schema, schemaValidator);

    this.setState({ schemaBridge });
  }

  ...

  render() {
    const { schemaBridge: schema } = this.state

    return(
    ...
    {
      schema !== null ?
      <AutoForm schema={schema} /> :
      <div />
    }
    ...
  }
}
```

</details>

<br/>

<details>

<summary>Console logs</summary>

```
The above error occurred in the <AutoValidatedQuickForm> component:
  in AutoValidatedQuickForm (at ReportModal.js:70)
  in div (created by Paper)
  in Paper (created by WithStyles(Paper))
  in WithStyles(Paper) (created by Dialog)
  in div (created by Dialog)
  in Transition (created by Fade)
  in Fade (created by WithTheme(Fade))
  in WithTheme(Fade) (created by Dialog)
  in RootRef (created by Modal)
  in div (created by Modal)
  in Portal (created by Modal)
  in Modal (created by WithStyles(Modal))
  in WithStyles(Modal) (created by Dialog)
  in Dialog (created by WithStyles(Dialog))
  in WithStyles(Dialog) (at ReportModal.js:61)
  in div (at ReportModal.js:53)
  in _default (at Reports.js:49)
  in div (created by Grid)
  in Grid (created by WithStyles(Grid))
  in WithStyles(Grid) (at Reports.js:48)
  in div (created by Grid)
  in Grid (created by WithStyles(Grid))
  in WithStyles(Grid) (at Reports.js:44)
  in _default (at App.js:12)
  in div (at App.js:10)
  in App (at src/index.js:8)

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://fb.me/react-error-boundaries to learn more about error boundaries. index.js:1446

TypeError: can't convert undefined to object  JSONSchemaBridge.js:276
  getSubfields  JSONSchemaBridge.js:276
  render  QuickForm.js:64
  finishClassComponent  react-dom.development.js:15141
  updateClassComponent  react-dom.development.js:15096
  beginWork  react-dom.development.js:15980
  performUnitOfWork  react-dom.development.js:19102
  workLoop  react-dom.development.js:19143
  callCallback  react-dom.development.js:147
  invokeGuardedCallbackDev  react-dom.development.js:196
  invokeGuardedCallback  react-dom.development.js:250
  replayUnitOfWork  react-dom.development.js:18350
  renderRoot  react-dom.development.js:19261
  performWorkOnRoot  react-dom.development.js:20165
  performWork  react-dom.development.js:20075
  performSyncWork  react-dom.development.js:20049
  interactiveUpdates$1  react-dom.development.js:20337
  interactiveUpdates  react-dom.development.js:2267
  dispatchInteractiveEvent  react-dom.development.js:5083
  dispatchInteractiveEvent self-hosted:1018:17
```

</details>

### PR DESCRIPTION
**Pull request by janowsiany**
Fixes #507

### CODE REVIEW

File: packages/uniforms/__tests__/JSONSchemaBridge.js [hunk 1]
```diff
@@ -1,87 +1,97 @@
 import JSONSchemaBridge from 'uniforms/JSONSchemaBridge';
 
+let schema;
+let validator;
+let bridge;
```
Comment (radekmie) [inline packages/uniforms/__tests__/JSONSchemaBridge.js]: Let's keep the `const`s and construct new values in these few tests.

File: packages/uniforms/src/JSONSchemaBridge.js [hunk 2]
```diff
@@ -26,6 +26,7 @@ export default class JSONSchemaBridge extends Bridge {
 
     this.schema = schema;
     this._compiledSchema = {};
+    this._rootSchema = this._getRootSchema();
```
Comment (radekmie) [inline packages/uniforms/src/JSONSchemaBridge.js]: I don't like this idea. How about resolving within `this.schema` it to actually keep only that one field? Like, if it's an object: done; if it's a `$ref` object, then construct a new object: `{...schema, ...resolveRef(schema.$ref, schema)}`. It'll work, right?

Comment (janowsiany) [reply packages/uniforms/src/JSONSchemaBridge.js]: My intention was to keep schema as is, but i can think about other solution

File: packages/uniforms/src/JSONSchemaBridge.js [hunk 3]
```diff
@@ -215,4 +219,16 @@ export default class JSONSchemaBridge extends Bridge {
...
+    if (schema.type === 'object') {
+      return schema;
+    }
+
+    if (schema.$ref) {
+      return {...schema, ...resolveRef(schema.$ref, schema)};
+    }
+
+    return schema;
+  }
```
Comment (radekmie) [inline packages/uniforms/src/JSONSchemaBridge.js]: This doesn't have to be a part of the class.

Comment (janowsiany) [reply packages/uniforms/src/JSONSchemaBridge.js]: What do you mean?

Comment (radekmie) [reply packages/uniforms/src/JSONSchemaBridge.js]: This could be extracted out of the class, same like the `resolveRef` function.

**Pull request merged**